### DISCUSSION TITLE
How should I do with `vercel dev` to test my serverless functions

### DISCUSSION THREAD

**Discussion by Zeng1998** [vercel/vercel#6593] [Help] [answered]
I have a vue3 project with some serverless functions, it is deployed in vercel successful and I can visit "xxxxxx.vercel.app/api" in browser to fetch the data of the API.

But if I run "vercel dev" locally, I can visit localhost:3001 to see my front-end, but if I visit localhost:3001/api,  it just like be handle by the front-end router and no data fetched.

I am confuse, maybe I made a mistake about that.

I used the vue-router in history mode.

**Comment by leerob** [comment, marked as answer]
Hey! Are you using the default [Vue template](https://vercel.com/new/clone?s=https%3A%2F%2Fgithub.com%2Fvercel%2Fvercel%2Ftree%2Fmain%2Fexamples%2Fvue&template=vue&id=67753070&b=main&from=templates)? Is Vue selected as the framework in your [project settings](https://vercel.com/docs/build-step#framework-preset)?
<img width="835" alt="CleanShot 2021-08-16 at 16 25 18@2x" src="https://user-images.githubusercontent.com/9113740/129631479-05d6eea4-e57d-4dd0-901a-31a8392b5aff.png">


You should be able to run `vercel dev` and start your Vue app as well as test your functions locally.

**Reply by Zeng1998** [reply]
I don't really understand.
I use `Vite` in my project. I want to know when I run `vercel dev` in my local machine, how can I access the serverless function api like `{domain}/api/xxxx`.  By now, when I enter  `{domain}/api/xxxx`. in browser, it just like be handled by the vue router.

**Reply by leerob** [reply]
In that case, you should likely have the framework set as "Vite" instead. You can also start from this example: https://github.com/vercel/vercel/tree/main/examples/vite

Let me know if that's not working for you!
